prefect_great_expectations/validation.py

```diff
@@ -0,0 +1,102 @@
+"""Tasks for performing Great Expectations validations"""
+
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+from great_expectations import DataContext
+from great_expectations.checkpoint import Checkpoint
+from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+from prefect import get_run_logger, task
+
+
+class GreatExpectationValidationError(Exception):
+    """
+    Signals that a task failed due to a failed Great
+    Expectations validation.
+
+    Args:
+        result: A CheckpointResult containing details
+            of the failed validation.
+    """
+
+    def __init__(self, result: CheckpointResult):
+        self.result = result
+        super().__init__(
+            "Great Expectations Validation failed. "
+            "Check result on this exception for more details."
+        )
+
+
+@task
+def run_checkpoint_validation(
+    run_name: Optional[str] = None,
+    checkpoint_name: Optional[str] = None,
+    checkpoint: Optional[Checkpoint] = None,
+    checkpoint_kwargs: Optional[Dict] = None,
+    data_context_root_dir: Optional[Union[str, Path]] = None,
+    data_context: Optional[DataContext] = None,
+    runtime_environment: Optional[Dict] = None,
+    raise_on_validation_failure: bool = True,
+):
+    """
+    Task that performs a Great Expectations validation based on the provided checkpoint
+        and data context
+
+    Args:
+        run_name: The name of the Great Expectations validation run. Defaults to
+            timestamp if not provided.
+        checkpoint_name: The name of the Checkpoint to use for validation.
+        checkpoint: A Checkpoint object to use for validation. Overrides
+            `checkpoint_name` if both are provided.
+        checkpoint_kwargs:  A dictionary with values used to provide configuration to
```

Comment:
```suggestion
        checkpoint_kwargs: A dictionary with values used to provide configuration to
```

Reply:
Should this just be `**checkpoint_kwargs`?

Reply:
There are two different objects that we interact with in this task: a `DataContext` and a `Checkpoint`. The analogous arg for `DataContext` is runtime environment. I prefer to have `checkpoint_kwargs` as an explicit arg so that there's no confusion when reading a task invocation where the additional args are going. If you looked just at an invocation of this task you wouldn't be able to tell if the extra args are going to the `DataContext` or the `Checkpoint`.